tests/simulator/diagnostics.py

```diff
@@ -156,18 +172,68 @@ def _test_dump_diags(self, dim, **simInput):
 
 
     @data(*_test_cases)
-    def test_dump_diags_1d(self, simInput):
-        self._test_dump_diags(1, **simInput)
+    def test_dump_diags(self, simInput):
+        for dim in [1, 2]:
+            self._test_dump_diags(dim, **simInput)
 
+    def _test_patch_ghost_on_refined_level_case(self, has_patch_ghost, **simInput):
+        test_id = self.ddt_test_id()
+        if cpp.mpi_size() > 1: # SKIP
+            return
 
-    @data(*_test_cases)
-    def test_dump_diags_2d(self, simInput):
-        self._test_dump_diags(2, **simInput)
+        for dim in [1, 2]:
+            b0 = [[10 for i in range(dim)], [20 for i in range(dim)]]
+            simInput["refinement_boxes"] = {"L0": {"B0": b0}}
+            for interp in [1, 2, 3]:
 
+                local_out = f"{out}_dim{dim}_interp{interp}_mpi_n_{cpp.mpi_size()}_id{test_id}"
+                simInput["diag_options"]["options"]["dir"] = local_out
+
+                simulation = ph.Simulation(**simInput)
+                dump_all_diags(setup_model().populations)
+                self.simulator = Simulator(simulation).initialize().advance()
+
+                for diagInfo in ph.global_vars.sim.diagnostics:
+                    print("diagInfo.quantity", diagInfo.quantity)
+
+                self.assertTrue(any([diagInfo.quantity.endswith("patchGhost") for diagInfo in ph.global_vars.sim.diagnostics]))
+
+                for diagInfo in ph.global_vars.sim.diagnostics:
+                    if diagInfo.quantity.endswith("patchGhost"):
+                        # diagInfo.quantity starts with a / this interferes with os.path.join, hence   [1:]
+                        h5_filename = os.path.join(local_out, (diagInfo.quantity + ".h5").replace('/', '_')[1:])
+                        h5_file = h5py.File(h5_filename, "r")
+                        if has_patch_ghost:
+                            self.assertTrue("weight" in h5_file["t0.000000"]["pl1"]["p0#0"])
+                        else:
+                            self.assertFalse("weight" in h5_file["t0.000000"]["pl1"]["p0#0"])
+
+                self.simulator = None
+                ph.global_vars.sim = None
+
+
+
+
+    _no_patch_ghost_on_refined_level_case = (
+      dup({
+        "smallest_patch_size": 40,
+        "largest_patch_size": 40}),
+    )
+    @data(*_no_patch_ghost_on_refined_level_case)
+    def test_no_patch_ghost_on_refined_level_case(self, simInput):
+        self._test_patch_ghost_on_refined_level_case(False, **simInput)
+
+
+
+    _has_patch_ghost_on_refined_level_case = (
+      dup({
+        "smallest_patch_size": 5,
+        "largest_patch_size": 5}),
+    )
+    @data(*_has_patch_ghost_on_refined_level_case)
+    def test_has_patch_ghost_on_refined_level_case(self, simInput):
```

Comment:
aren't these init tests ?

Reply:
we're doing direct h5py access, so maybe not, or maybe there's another way via the hierarchy

Reply:
moved to test_initlization.py